modules/cockroachdb/cockroachdb_test.go

```diff
@@ -144,6 +147,71 @@ func (suite *AuthNSuite) TestQuery() {
 	suite.Equal(523123, id)
 }
 
+// TestWithWaitStrategyAndDeadline covers a previous regression, container creation needs to fail to cover that path.
+func (suite *AuthNSuite) TestWithWaitStrategyAndDeadline() {
+
+	contextDeadlineExceeded := fmt.Errorf("failed to start container: context deadline exceeded")
+	nodeStartUpCompleted := "node startup completed"
+	suite.Run("Expected Failure To Run", func() {
+
+		ctx := context.Background()
+
+		// This will never match a log statement
+		suite.opts = append(suite.opts, testcontainers.WithWaitStrategyAndDeadline(time.Millisecond*250, wait.ForLog("Won't Exist In Logs")))
+		container, err := cockroachdb.RunContainer(ctx, suite.opts...)
+		suite.Require().Error(err, contextDeadlineExceeded)
+		suite.T().Cleanup(func() {
+			if container != nil {
+				err := container.Terminate(ctx)
+				suite.Require().NoError(err)
+			}
+		})
+
+	})
+
+	suite.Run("Expected Failure To Run But Would Succeed ", func() {
+
+		ctx := context.Background()
+
+		// This will timeout as we didn't give enough time for intialization, but would have succeeded otherwise
+		suite.opts = append(suite.opts, testcontainers.WithWaitStrategyAndDeadline(time.Millisecond*20, wait.ForLog(nodeStartUpCompleted)))
+		container, err := cockroachdb.RunContainer(ctx, suite.opts...)
+		suite.Require().Error(err, contextDeadlineExceeded)
+		suite.T().Cleanup(func() {
+			if container != nil {
+				err := container.Terminate(ctx)
+				suite.Require().NoError(err)
+			}
+		})
+
+	})
+
+	suite.Run("Succeeds And Executes Commands", func() {
+
+		ctx := context.Background()
+
+		// This will succeed
+		suite.opts = append(suite.opts, testcontainers.WithWaitStrategyAndDeadline(time.Second*60, wait.ForLog(nodeStartUpCompleted)))
+		container, err := cockroachdb.RunContainer(ctx, suite.opts...)
+		suite.Require().NoError(err)
+
+		conn, err := conn(ctx, container)
+		suite.Require().NoError(err)
+		defer conn.Close(ctx)
+
+		_, err = conn.Exec(ctx, "CREATE TABLE test (id INT PRIMARY KEY)")
+		suite.Require().NoError(err)
+		suite.T().Cleanup(func() {
+			if container != nil {
+				err := container.Terminate(ctx)
+				suite.Require().NoError(err)
+			}
+		})
+
+	})
```

Comment:
I notice that there is inconsistency between how we are closing the db connection ("defer") and how we are terminating the container (using the cleanup functionality). Is there a reason for this? Can't we just use one?

Reply:
I'm not the original maintainer of this. This is what the tests were doing previously, it's also whats done project wide. See for example - https://github.com/testcontainers/testcontainers-go/blob/main/modules/postgres/postgres_test.go